src/api/app/models/package.rb

```diff
@@ -357,8 +359,12 @@ def kiwi_image_outdated?
 
   def master_product_object
     # test _product permissions if any other _product: subcontainer is used and _product exists
-    product_object = project.packages.find_by(name: '_product') if name =~ /\A_product:\w[-+\w\.]*\z/
-    product_object || self
+    return self unless belongs_to_product?
+    project.packages.product.first
```

Comment:
this would return a product object instead of a package object, right?

project.packages.product.first.package

Reply:
Oh, right. But in this case we want the package that contains a product, not the product object.

I am going to rename the scope to avoid confusion.